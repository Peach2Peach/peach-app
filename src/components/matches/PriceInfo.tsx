import React from 'react'
import { View } from 'react-native'
import shallow from 'zustand/shallow'
import tw from '../../styles/tailwind'
import i18n from '../../utils/i18n'
import { getPaymentMethodInfo } from '../../utils/paymentMethod'
import { padString } from '../../utils/string'
import { Text } from '../text'
import { HorizontalLine } from '../ui'
import { useMatchStore } from './store'

const getDisplayPrice = (match: Match, selectedPaymentMethod: PaymentMethod, selectedCurrency: Currency) => {
  const paymentInfo = getPaymentMethodInfo(selectedPaymentMethod)
  let displayPrice = String(
    match.matched && match.matchedPrice
      ? match.matchedPrice
      : paymentInfo?.rounded
        ? Math.round(match.prices[selectedCurrency]!)
        : match.prices[selectedCurrency],
  )
  displayPrice = `${displayPrice.split('.')[0]}.${padString({
    string: displayPrice.split('.')[1],
    length: 2,
    char: '0',
    side: 'right',
  })}`

  return displayPrice
}

type PriceInfoProps = {
  match: Match
}

export default ({ match }: PriceInfoProps) => {
  const { selectedCurrency, selectedPaymentMethod } = useMatchStore(
    (state) => ({ selectedCurrency: state.selectedCurrency, selectedPaymentMethod: state.selectedPaymentMethod }),
    shallow,
  )
  const displayPrice = getDisplayPrice(match, selectedPaymentMethod, selectedCurrency)
  return (
    <>
      <HorizontalLine style={[tw`mt-3`, tw.md`mt-4`]} />
      <View style={[tw`mt-3`, tw.md`mt-4`]}>
        <Text style={tw`font-baloo text-xl leading-xl text-peach-1 text-center`}>
          {i18n(`currency.format.${selectedCurrency}`, displayPrice)}
        </Text>
        <Text style={tw`text-lg leading-lg text-center text-grey-2 ml-2`}>
          (
          {i18n(match.premium > 0 ? 'offer.summary.premium' : 'offer.summary.discount', String(Math.abs(match.premium)))}
          )
        </Text>
      </View>
    </>
  )
}
