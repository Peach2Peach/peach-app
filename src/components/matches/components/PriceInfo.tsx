import { View } from "react-native";
import { Currency } from "../../../../peach-api/src/@types/global";
import { SATSINBTC } from "../../../constants";
import tw from "../../../styles/tailwind";
import { BTCAmount } from "../../bitcoin/BTCAmount";
import { PeachText } from "../../text/PeachText";
import { PriceFormat } from "../../text/PriceFormat";
import { PremiumText } from "./PremiumText";
import { useTranslate } from "@tolgee/react";

type Props = {
  amount: number;
  price: number;
  currency: Currency;
  premium: number;
};
export function PriceInfo({ amount, price, currency, premium }: Props) {
  const btcPrice = Math.round((price / amount) * SATSINBTC);
  const { t } = useTranslate("global");
  return (
    <View style={tw`items-center`}>
      <View style={tw`items-center justify-center h-7`}>
        <BTCAmount amount={amount} size="medium" />
      </View>
      <PeachText style={tw`text-center`}>
        <PriceFormat
          style={tw`subtitle-1`}
          currency={currency}
          amount={price}
        />
        <PremiumText premium={premium} />
      </PeachText>
      <PeachText style={tw`subtitle-1`}>
        {btcPrice} {t(currency)} / {t("btc")} {/* TODO: understand and fix*/}
      </PeachText>
    </View>
  );
}
