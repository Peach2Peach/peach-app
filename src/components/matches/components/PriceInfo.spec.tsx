import { createRenderer } from 'react-test-renderer/shallow'
import { buyOffer, matchOffer } from '../../../../tests/unit/data/offerData'
import { PriceInfo } from './PriceInfo'

jest.mock('../../../hooks', () => ({
  useMarketPrices: jest.fn(() => ({
    data: {
      EUR: 400,
    },
    isSuccess: true,
  })),
}))

describe('PriceInfo', () => {
  it('should render correctly', () => {
    const renderer = createRenderer()
    renderer.render(<PriceInfo match={matchOffer} offer={buyOffer} />)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
