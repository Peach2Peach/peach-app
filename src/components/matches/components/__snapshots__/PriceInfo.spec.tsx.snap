// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceInfo should render correctly 1`] = `
<React.Fragment>
  <SatsFormat
    bitcoinLogoStyle={
      {
        "height": 18,
        "marginRight": 8,
        "width": 18,
      }
    }
    containerStyle={
      {
        "alignSelf": "center",
        "justifyContent": "center",
      }
    }
    sats={210000}
    satsStyle={
      {
        "fontFamily": "Baloo2-SemiBold",
        "fontSize": 16,
        "letterSpacing": 0.16,
        "lineHeight": 28,
      }
    }
    style={
      {
        "fontFamily": "Baloo2-Bold",
        "fontSize": 24,
        "letterSpacing": 0.24,
        "lineHeight": 42,
      }
    }
  />
  <PeachText
    style={
      {
        "textAlign": "center",
      }
    }
  >
    <Price
      match={
        {
          "amount": 210000,
          "premium": 7,
        }
      }
      offer={{}}
      textStyle={
        {
          "fontFamily": "Baloo2-SemiBold",
          "fontSize": 16,
          "letterSpacing": 0.16,
          "lineHeight": 28,
        }
      }
    />
    <PremiumText
      premium={7}
    />
  </PeachText>
</React.Fragment>
`;
